sectorbuilder.go

```diff
@@ -388,3 +416,60 @@ func (sb *SectorBuilder) SetLastSectorNum(num abi.SectorNumber) error {
 func (sb *SectorBuilder) Stop() {
 	close(sb.stopping)
 }
+
+func sizeFromConfig(cfg Config) (abi.SectorSize, error) {
+	if cfg.SealProofType == abi.RegisteredProof(0) {
+		return abi.SectorSize(0), xerrors.New("must specify a seal proof type from abi.RegisteredProof")
+	}
+
+	if cfg.PoStProofType == abi.RegisteredProof(0) {
+		return abi.SectorSize(0), xerrors.New("must specify a PoSt proof type from abi.RegisteredProof")
+	}
+
+	s1, err := sizeFromProofType(cfg.SealProofType)
+	if err != nil {
+		return abi.SectorSize(0), err
+	}
+
+	s2, err := sizeFromProofType(cfg.PoStProofType)
+	if err != nil {
+		return abi.SectorSize(0), err
+	}
+
+	if s1 != s2 {
+		return abi.SectorSize(0), xerrors.Errorf("seal sector size %d does not equal PoSt sector size %d", s1, s2)
+	}
+
+	return s1, nil
+}
+
+func sizeFromProofType(p abi.RegisteredProof) (abi.SectorSize, error) {
+	switch p {
+	case abi.RegisteredProof_WinStackedDRG32GiBSeal:
+		return 1 << 35, nil
```

Comment:
It would be better if this mapping lived with proofs (probably not easily achievable). If sectorbuilder is going to own it, it should expose `sizeFromProofType` to the implementations so they don't need to re-implement this map.